assets/javascripts/discourse/views/select2.js.es6

```diff
@@ -0,0 +1,24 @@
+export default Ember.Select.reopen({
+
+  didInsertElement: function() {
+    Ember.run.scheduleOnce('afterRender', this, 'processChildElements');
+  },
+
+  processChildElements: function() {
+    var options = {
+      allowClear: true,
+      closeOnSelect: true,
+      width: '200px'
+    };
+
+    this.$().select2(options);
+  },
+
+  willDestroyElement: function () {
+    this.$().select2('destroy');
+  },
+
+  selectedDidChange: function() {
```

Comment:
You can use this ember functionality https://guides.emberjs.com/v1.10.0/object-model/observers/#toc_without-prototype-extensions.